packages/web-shared/src/sidebar/attribute-panel.tsx

```diff
@@ -123,9 +122,68 @@ const attributeToDisplayFn: Record<
     return <DetailCard summary="Output">{JsonBlock(value)}</DetailCard>;
   },
   error: (value: unknown) => {
-    return <DetailCard summary="Error">{JsonBlock(value)}</DetailCard>;
+    // Handle structured error format
+    if (value && typeof value === 'object' && 'message' in value) {
+      const error = value as {
+        message: string;
+        stack?: string;
+        code?: string;
+      };
+
+      return (
+        <DetailCard summary="Error">
+          <div className="flex flex-col gap-2">
+            {/* Show code if it exists */}
+            {error.code && (
+              <div>
+                <span
+                  className="text-copy-12 font-medium"
+                  style={{ color: 'var(--ds-gray-700)' }}
+                >
+                  Error Code:{' '}
+                </span>
+                <code
+                  className="text-copy-12"
+                  style={{ color: 'var(--ds-gray-1000)' }}
+                >
+                  {error.code}
+                </code>
+              </div>
+            )}
+            {/* Show stack if available, otherwise just the message */}
+            <pre
+              className="text-copy-12 overflow-x-auto rounded-md border p-4"
+              style={{
+                borderColor: 'var(--ds-gray-300)',
```

Comment:
Not relevant to PR, but reminder to myself that I gotta fix these variable names to be tw classes